connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -185,51 +224,45 @@ public void poll(long timeoutMs) {
      * the write commit. This should only be invoked by the WorkerSinkTaskThread.
      **/
-    public void commitOffsets(boolean sync, final int seqno) {
+    private void commitOffsets(Map<TopicPartition, OffsetAndMetadata> offsets, boolean sync, final int seqno) {
         log.info("{} Committing offsets", this);
-
-        final Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>(currentOffsets);
-
-        try {
-            task.flush(offsets);
-        } catch (Throwable t) {
-            log.error("Commit of {} offsets failed due to exception while flushing:", this, t);
-            log.error("Rewinding offsets to last committed offsets");
-            for (Map.Entry<TopicPartition, OffsetAndMetadata> entry : lastCommittedOffsets.entrySet()) {
-                log.debug("{} Rewinding topic partition {} to offset {}", id, entry.getKey(), entry.getValue().offset());
-                consumer.seek(entry.getKey(), entry.getValue().offset());
-            }
-            currentOffsets = new HashMap<>(lastCommittedOffsets);
-            workThread.onCommitCompleted(t, seqno);
-            return;
-        }
-
         if (sync) {
             try {
                 consumer.commitSync(offsets);
                 lastCommittedOffsets = offsets;
-                workThread.onCommitCompleted(null, seqno);
+                onCommitCompleted(null, seqno);
             } catch (KafkaException e) {
-                workThread.onCommitCompleted(e, seqno);
+                onCommitCompleted(e, seqno);
             }
         } else {
             OffsetCommitCallback cb = new OffsetCommitCallback() {
                 @Override
                 public void onComplete(Map<TopicPartition, OffsetAndMetadata> offsets, Exception error) {
                     lastCommittedOffsets = offsets;
-                    workThread.onCommitCompleted(error, seqno);
+                    onCommitCompleted(error, seqno);
                 }
             };
             consumer.commitAsync(offsets, cb);
         }
     }
 
-    public Time time() {
-        return time;
-    }
+    private void flushAndCommitOffsets(boolean sync, int seqno) {
+        Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>(currentOffsets);
+        try {
+            task.flush(offsets);
+        } catch (Throwable t) {
+            log.error("Commit of {} offsets failed due to exception while flushing:", this, t);
+            log.error("Rewinding offsets to last committed offsets");
+            for (Map.Entry<TopicPartition, OffsetAndMetadata> entry : lastCommittedOffsets.entrySet()) {
+                log.debug("{} Rewinding topic partition {} to offset {}", id, entry.getKey(), entry.getValue().offset());
+                consumer.seek(entry.getKey(), entry.getValue().offset());
+            }
+            currentOffsets = new HashMap<>(lastCommittedOffsets);
+            onCommitCompleted(t, seqno);
```

Comment:
I agree it's a little confusing. It was clearer before when the name of the method was just commitOffsets, but I had to split flushing and committing to avoid the flush() call after closing the partitions. Let me see if there's a way to restructure the code to make this nicer.